DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc

```diff
@@ -0,0 +1,1234 @@
+
+	  objEtHistEmul->Fill(emulIt->hwPt());
+	  objEtaHistEmul->Fill(emulIt->hwEta());
+	  objPhiHistEmul->Fill(emulIt->hwPhi());
+	}
+
+	problemSummary->getTH1F()->Fill(TAUMISMATCH);
+      }
+
+      // if only position agrees
+      if (posGood && !etGood) {
+	if (iso) {
+	  objSummaryHist->Fill(ISOTAUETOFF);
+	} else {
+	  objSummaryHist->Fill(TAUETOFF);
+	}
+      }
+
+      // if only energy agrees
+      if (!posGood && etGood) {
+	if (iso) {
+	  objSummaryHist->Fill(ISOTAUPOSOFF);
+	} else {
+	  objSummaryHist->Fill(TAUPOSOFF);
+	}
+      }
+
+      // keep track of number of objects
+      if (iso) {
+	objSummaryHist->Fill(NISOTAUS);
+      } else {
+	objSummaryHist->Fill(NTAUS);
+      }
+
+      summaryHist->Fill(NTAUS_S);
+
+      // increment position of pointers
+      ++dataIt;
+      ++emulIt;
+
+      if (dataIt == dataCol->end(currBx) ||
+	  emulIt == emulCol->end(currBx))
+	break;
+    }
+  } else {
+    return false;
+  }
+
+  // return a boolean that states whether the jet data in the event is in
+  // agreement
+  return eventGood;
+}
+
+// comparison method for sums
+bool L1TdeStage2CaloLayer2::compareSums(
+  const edm::Handle<l1t::EtSumBxCollection> & dataCol,
+  const edm::Handle<l1t::EtSumBxCollection> & emulCol)
+{
+  bool eventGood = true;
+
+  bool etGood = true;
+  bool phiGood = true;
+
+  l1t::EtSumBxCollection::const_iterator dataIt = dataCol->begin(currBx);
+  l1t::EtSumBxCollection::const_iterator emulIt = emulCol->begin(currBx);
+
+  TH1F * objSummaryHist = sumSummary->getTH1F();
+  TH1F * summaryHist = agreementSummary->getTH1F();
+  TH1F * problemHist = problemSummary->getTH1F();
+
+  while(true) {
+
+    // It should be possible to implement this with a switch statement
+    etGood = true;
+    phiGood = true;
+
+    // ETT
+    if (l1t::EtSum::EtSumType::kTotalEt == dataIt->getType()) {
+
+      objSummaryHist->Fill(NETTSUMS);
+      summaryHist->Fill(NSUMS_S);
+
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+	eventGood = false;
+
+	ettData->getTH1F()->Fill(dataIt->hwPt());
+	ettEmul->getTH1F()->Fill(emulIt->hwPt());
+
+      } else {
+	summaryHist->Fill(SUMGOOD_S);
+	objSummaryHist->Fill(SUMGOOD);
+	objSummaryHist->Fill(ETTSUMGOOD);
+      }
+    }
+
+    // ETTHF
+    if (l1t::EtSum::EtSumType::kTotalEtHF == dataIt->getType()) {
+
+      objSummaryHist->Fill(NETTSUMS);
+      summaryHist->Fill(NSUMS_S);
+
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+	eventGood = false;
+
+	ettHFData->getTH1F()->Fill(dataIt->hwPt());
+	ettHFEmul->getTH1F()->Fill(emulIt->hwPt());
+
+      } else {
+	summaryHist->Fill(SUMGOOD_S);
+	objSummaryHist->Fill(SUMGOOD);
+	objSummaryHist->Fill(ETTSUMGOOD);
+      }
+    }
+
+    // ETTEM
+    if (l1t::EtSum::EtSumType::kTotalEtEm == dataIt->getType()) {
+
+      objSummaryHist->Fill(NETTSUMS);
+      summaryHist->Fill(NSUMS_S);
+
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+	eventGood = false;
+	ettEmData->getTH1F()->Fill(dataIt->hwPt());
```

Comment:
essentially none of the getTH1F() are needed - the MonitorElement has a Fill method to call directly.